src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketWebhookClientImplTest.java

```diff
@@ -87,4 +93,35 @@ public void testRegisterWebhook() {
 
         assertThat(result.getEvents(), hasItems(repoRefEvent, mirrorSyncEvent));
     }
+
+    @Test
+    public void testNextPageFetching() {
+        NextPageFetcherImpl fetcher = new NextPageFetcherImpl(parse(BITBUCKET_BASE_URL), bitbucketRequestExecutor);
+        int nextPageStart = 2;
+        fakeRemoteHttpServer.mapUrlToResult(
+                BITBUCKET_BASE_URL + "?start=" + nextPageStart,
+                readFileToString("/webhook/web_hooks_in_system_last_page.json"));
+        BitbucketPage<BitbucketWebhook> firstPage = new BitbucketPage<>();
+        firstPage.setNextPageStart(nextPageStart);
+
+        BitbucketPage<BitbucketWebhook> next = fetcher.next(firstPage);
+        assertEquals(next.getSize(), next.getValues().size());
+        assertTrue(next.getSize() > 0);
+
+        BitbucketWebhook webhook = next.getValues().get(0);
+        assertNotNull(webhook.getName());
+        assertTrue(webhook.getId() > 0);
```

Comment:
maybe assert the actual id? I know this was my suggestion but in retrospect it is a bit basic we can do better